### ISSUE TITLE
False positive ParadoxicalCondition caused by old assertion even when variable is overwritten

### ISSUE THREAD

**Issue by TysonAndre** [vimeo/psalm#231]
Expected: no error

Observed:
```
ERROR: ParadoxicalCondition - src/leftover.php:13:9 - Encountered a paradox when evaluating the conditional
    if ($res === false) {
```

```php
<?php

function get_bool() : bool {
    return rand() % 2 > 0;
}

function leftover() : bool {
    $res = get_bool();
    if ($res === false) {
        return true;
    }   
    $res = get_bool();
    if ($res === false) {  // This is a false positive
        return false;
    }   
    return true;
}
```

**Comment by muglug** [comment]
Thanks! This was due to the special handling of `false`, so I've added a specific workaround.
